### DISCUSSION TITLE
having trouble using ImageMagick with moviepy

### DISCUSSION THREAD

**Discussion by Izolovi4** [ImageMagick/ImageMagick#7586] [Help]
### ImageMagick version

7.1.1-Q16-HDRI

### Operating system, version and so on

windows 10, python 3.12, moviepy 1.0.3

### Description

I was using it to add a new font and got these errors:

> magick.exe: UnableToOpenConfigureFile `colors.xml' @ warning/configure.c/GetConfigureOptions/722.

> magick.exe: UnableToOpenBlob 'label:@C:\Users\asdf\AppData\Local\Temp\tmp19qk5f31.txt': Invalid argument @ error/blob.c/OpenBlob/3596.

> magick.exe: UnableToOpenModuleFile 'C:\Program Files\ImageMagick-7.1.1-Q16\modules\coders\IM_MOD_RL_TXT_.dll': No such file or directory @ warning/module.c/GetMagickModulePath/678.

> magick.exe: NoDecodeDelegateForThisImageFormat `TXT' @ error/constitute.c/ReadImage/746.

The weirdest of 'em all is the third one, because it is checking a version, that doesn't even exist

**Comment by fmw42** [comment]
7.1.1-Q16-HDRI is not a valid version.  It should be 7.1.1.x-Q16-HDRI.  Please identify the version of Imagemagick that you installed and how you installed it. From where did you get it?  (try magick -version).  Is Moviepy compatible with Imagemagick 7 or does it require Imagemagick 6?

**Reply by Izolovi4** [reply]
> is not valid version

just checked through terminal, it was 7.1.1-37 Q16-HDRI x64

>  Is Moviepy compatible with Imagemagick 7

yep

**Reply by fmw42** [reply]
How did you install Imagemagick 7?   From where did you get the install?

**Reply by fmw42** [reply]
What do you get from 

`magick -version`

**Reply by Izolovi4** [reply]
> How did you install Imagemagick 7? From where did you get the install?

![изображение](https://github.com/user-attachments/assets/f19e22cf-e2f8-4bd0-ac19-3ade956f4396)


> What do you get from
> 
> `magick -version`

Version: ImageMagick 7.1.1-37 Q16-HDRI x64 1ffe565:20240824 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Channel-masks(64-bit) Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (194033813)

**Reply by fmw42** [reply]
That does not tell me how you installed it and where it came from?

You are missing fontconfig delegate from your list of delegates. It is needed when using fonts.

**Reply by Izolovi4** [reply]
> That does not tell me how you installed it and where it came from?

from the official website https://imagemagick.org/script/download.php

> You are missing fontconfig delegate from your list of delegates. It is needed when using fonts.

where can i get it from?

**Reply by fmw42** [reply]
Did you install the correct version for your OS?  Did you search Google for fontconfig?  See https://www.freedesktop.org/wiki/Software/fontconfig/.  I do not know if that is your problem. I am not a Windows user so do not know why it is not in your install.  Seems odd to me.

**Comment by snibgo** [comment]
What was your IM command?

The first error is:
> magick.exe: UnableToOpenConfigureFile `colors.xml' @ warning/configure.c/GetConfigureOptions/722.

This means IM can't find its xml files. Did you install IM, and then move it to a different directory? If you want to move IM, you should install the portable version.

What is the text output (sent to stderr) from:
```
magick -debug all xc:Red NULL:
```

**Reply by fmw42** [reply]
He is running it from MoviePy (python) tool.
